### PR TITLE
Database functions added

### PR DESCRIPTION
**Pull request by MilkywayPwns**
This adds some functions related to SQLite :3

### CODE REVIEW

File: Source/Shared/Database.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	colstr[colstr.size() - 1] = '\0';
+	valstr[valstr.size() - 1] = '\0';
+
+	// Create SQL query string
+	std::string SQL = "INSERT INTO " + Database::EscapeString(table) + " (" + colstr + ") ";
+	"VALUES " + valstr;
+
+	// Execute SQL query
+	char *error;
+	int result = sqlite3_exec(Database::DatabasePtr, SQL.c_str(), nullptr, nullptr, &error);
```
Comment (Convery) [inline Source/Shared/Database.cpp]: Might want to use VAPrint to get the error logged.

File: Source/Shared/Database.cpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		bool firstEntry = true;
+		for (auto col : where_stmt)
+		{
+			data_where += ((firstEntry) ? "WHERE " : "AND ") + Database::EscapeString(col.first) + "='" + Database::EscapeString(col.second.Value()) + "' ";
+			firstEntry = false;
+		}
+	}
+
+	// Build SQL query
+	std::string SQL = "SELECT " + data_select + " "
```
Comment (Convery) [inline Source/Shared/Database.cpp]: While msvc merges strings, you might want a '+' to ensure compatibility with other systems.

File: Source/Shared/Database.cpp [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+bool Database::Load(std::string filepath)
+{
+	// Open the database connection, returns true if succeeded
+	return (sqlite3_open(filepath.c_str(), &Database::DatabasePtr) == SQLITE_OK);
+}
+void Database::Save()
+{
+	// Close the database connection
+	sqlite3_close(Database::DatabasePtr);
+}
```
Comment (Convery) [inline Source/Shared/Database.cpp]: Missing newline at the end of the file, messes up some parsers.

File: Source/Shared/Database.cpp (hunk 2 above)
Comment (MilkywayPwns) [inline Source/Shared/Database.cpp]: Good to know for next time :P

**Pull request merged**